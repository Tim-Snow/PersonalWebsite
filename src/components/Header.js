import React from 'react'
import HeaderLink from './HeaderLink'
import HeaderTitle from './HeaderTitle'

const Header = () => (
    <div style={styles.container}>
        <HeaderTitle />
        <HeaderLink name="Portfolio" />
        <HeaderLink name="About" />
        <HeaderLink name="Contact" />
    </div>
)

const styles = {
    container: {
        display: 'flex',
        justifyContent: 'stretch',
        alignItems: 'center',
        position: 'fixed',
        height: '32px',
        top: 0,
        width: '100%',
        border: '1px solid black',
        backgroundColor: '#ddd',
    },
}

export default Header
