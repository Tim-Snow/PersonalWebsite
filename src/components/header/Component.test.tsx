import * as React from 'react'
import * as Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

import Header from './Component'

configure({ adapter: new Adapter() })

const defaultProps = {
  normal: () => undefined,
  game: () => undefined,
  about: () => undefined,
  contact: () => undefined,
  portfolio: () => undefined,
  type: 'reg',
  view: 'about',
}

describe('header', () => {
  beforeEach(() => {
    window.addEventListener = jest.fn()
  })

  it('should match the snapshot', () => {
    const header = shallow(<Header {...defaultProps} />)
    expect(header).toMatchSnapshot()
  })
})
