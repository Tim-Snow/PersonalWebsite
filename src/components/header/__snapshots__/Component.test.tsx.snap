// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    about={[Function]}
    contact={[Function]}
    portfolio={[Function]}
    toggle={[Function]}
    type="reg"
    view="about"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          style={
            Object {
              "flex": 1,
            }
          }
        >
          Tim Snow.
        </h2>,
        <Connect(Hamburger) />,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#007",
        "color": "#FFF",
        "display": "flex",
        "height": "50px",
        "left": 0,
        "marginBottom": "20px",
        "padding": "0px",
        "top": 0,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Tim Snow.",
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": "Tim Snow.",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            style={
              Object {
                "flex": 1,
              }
            }
          >
            Tim Snow.
          </h2>,
          <Connect(Hamburger) />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#007",
          "color": "#FFF",
          "display": "flex",
          "height": "50px",
          "left": 0,
          "marginBottom": "20px",
          "padding": "0px",
          "top": 0,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tim Snow.",
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": "Tim Snow.",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
